### ISSUE TITLE
Impossible to have favorite atom

### ISSUE THREAD

**Issue by gi94** [micheleg/dash-to-dock#676]
Im using Debian 9.3, gnome 3.22, dash to dock v61. 

Install atom using snap. The option to make atom favorite does not appear...

Dash to dock dont have support to snap?

![bug](https://user-images.githubusercontent.com/35580864/35124120-3a6629be-fca5-11e7-94fc-b26f1f488684.png)

**Comment by pdimh** [comment]
Hi @gi94,

I believe this is an issue of Gnome. Please, have a look at [how-do-i-add-eclipse-to-my-gnome-shell-favorites]( https://unix.stackexchange.com/questions/58824/how-do-i-add-eclipse-to-my-gnome-shell-favorites) (answer number 4)

**Comment by micheleg** [comment]
Sorry for the late reply. Yes, this is an upstream bug. I believe it is solved in Ubuntu, and probably upstream too, but not in the versions available in Debian (I have the same problem as you)

**Comment by skerit** [comment]
I had this issue with _Firefox Developer Edition_, a non-snap version of it.

The solution mentioned on StackOverflow didn't work, changing the name of the `.desktop` file to the binary file being executed did nothing.

What did work, was changing the name of the `.desktop` file to the `WM_CLASS` of the application.

In my case, I wanted a separate icon for Firefox & Firefox Developer Edition, so I added `--class=firefoxdev` to the `.desktop`'s `Exec` command, and then I named the file `firefoxdev.desktop`
